app/src/main/java/com/github/polypoly/app/base/game/Game.kt

```diff
@@ -0,0 +1,158 @@
+package com.github.polypoly.app.base.game
+
+import com.github.polypoly.app.base.PastGame
+import com.github.polypoly.app.base.game.location.InGameLocation
+import com.github.polypoly.app.base.game.location.LocalizationLevel
+import com.github.polypoly.app.base.game.rules_and_lobby.GameLobby
+import com.github.polypoly.app.base.game.rules_and_lobby.GameMode
+import com.github.polypoly.app.base.game.rules_and_lobby.GameRules
+import com.github.polypoly.app.base.user.User
+
+/**
+ * Represent the game and the current state of the game
+ * @property admin the [User] who is the admin of the game
+ * @property players the [Player]s of the game
+ * @property rules the rules of the [Game]
+ * @property inGameLocations the [InGameLocation]s of the [Game]
+ * @property currentRound the current round of the [Game]
+ * @property dateBegin the date and time when the [Game] has started in Unix time
+ * (seconds since 1970-01-01T00:00:00Z)
+ */
+class Game private constructor(
+    val admin: User = User(),
+    var players: List<Player> = listOf(),
+    val rules: GameRules = GameRules(),
+    val dateBegin: Long = System.currentTimeMillis(),
+) {
+
+    private val inGameLocations: List<InGameLocation> = rules.gameMap.flatMap { it.locations.map { location ->
```

Comment:
`it` shouldn't be used as is in lambdas (even if the fact that it is implicitly define is useful), but should instead be renamed with a more expressive name: `.flatMap { good_name ->  good_name.locations...`

Reply:
thank you, fix done